java.base/src/main/java/sun/nio/ch/Net$_native.java

```diff
@@ -264,53 +304,55 @@ private static InetAddress localInetAddress(FileDescriptor fd) throws IOExceptio
     }
 
     private static int getIntOption0(FileDescriptor fd, boolean mayNeedConversion, int level, int opt) throws IOException {
-        c_int result = auto();
-        struct_linger linger = auto();
-        c_char carg = auto();
-        void_ptr arg = auto(addr_of(result).cast());
-        socklen_t arglen = auto(sizeof(result)).cast();
+        // This code has more copy & paste duplicated logic than the original C version to
+        // workaround a qbicc bug that produces invalid llvm ir when we have block-args that contain addresses.
```

Comment:
Is this something that we've discussed already>

Reply:
I originally hit it the qbicc problem around the same time as the pointer/block arg errors covered by https://github.com/qbicc/qbicc/issues/1612.  I think when we fixed 1612, I assumed this was fixed too but never went back to verify that (just left the problem code block commented out with a throw of "Unreachable" because we weren't actually executing it).  

I'll see if I can pull it out into a separate test case and open an issue.  It has something to do with the pattern of having local variable that is a void_ptr which is conditionally assigned multiple times with the casted address of variables of different types.

I can also pull this particular part of the change into a separate PR to make it easier to track & revert later.

Reply:
As long as the issue is reported I don't mind merging this to move forward. I just don't want to lose track of the problem.

Reply:
By moving the cast to the "right" place as discussed in https://github.com/qbicc/qbicc/issues/1675, we can write this code the "right" way...